api/subscriptions/message_cache.go

```diff
@@ -41,16 +41,16 @@ func newMessageCache[T any](cacheSize uint32) *messageCache[T] {
 // it will generate the message and add it to the cache. The second return value
 // indicates whether the message is newly generated.
 func (mc *messageCache[T]) GetOrAdd(id thor.Bytes32, createMessage func() (T, error)) (T, bool, error) {
-	mc.mu.RLock()
-	msg, ok := mc.cache.Get(id)
-	mc.mu.RUnlock()
+	mc.w.RLock()
+	msg, ok := mc.cache.Peek(id)
+	mc.w.RUnlock()
 	if ok {
 		return msg.(T), false, nil
 	}
 
-	mc.mu.Lock()
-	defer mc.mu.Unlock()
-	msg, ok = mc.cache.Get(id)
+	mc.w.Lock()
+	defer mc.w.Unlock()
+	msg, ok = mc.cache.Peek(id)
```

Comment:
is it possible to add some tests around this new method ? or the old ones will still be enough. As I understand we cant add the actual reproduce method - it is a bit tricky

Reply:
I don't think it's possible to create a test that will be giving errors 100% of the time, when there is an actual problem.